src/hooks/item.ts

```diff
@@ -97,8 +123,15 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
         enabled: enabled && Boolean(id),
       }),
 
-    useSharedItems: () =>
-      useQuery({
+    useSharedItems: (options?: { getUpdates?: boolean }) => {
+      const getUpdates = options?.getUpdates ?? enableWebsocket;
+
+      const { data: currentMember } = useCurrentMember();
+      itemWsHooks?.useSharedItemsUpdates(
+        getUpdates ? currentMember?.get('id') : null,
+      );
```

Comment:
Btw is there an issue here if the `currentMember` takes time to load and is `undefined` first for some time?

Reply:
I've wondered the same but I don't think it is an issue. `useCurrentMember` may return undefined as data first (and `isLoading` set to true) but the rest of the hook will still execute (in this case the shared items will still get fetched from the query client) and the WS hook will be called with null and do nothing. I believe when `useCurrentMember` finally returns the actual `currentMember`, the hook will be recalled and that time `getUpdates ? currentMember?.get('id')` will return the actual id and perform the WS subscription

Reply:
I think so as well, but I wanted to confirm with you^^